motion/thruster_allocator_auv/include/thruster_allocator_auv/thruster_allocator_utils.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file thruster_
+ * allocator_utils.hpp
+ * @brief This file contains utility functions for the thruster allocator
+ * module.
+ */
+
+#ifndef VORTEX_ALLOCATOR_ALLOCATOR_UTILS_HPP
+#define VORTEX_ALLOCATOR_ALLOCATOR_UTILS_HPP
+
+#include <eigen3/Eigen/Eigen>
+#include <rclcpp/rclcpp.hpp>
+#include <string>
+#include <vector>
+
+#include <vortex_msgs/msg/thruster_forces.hpp>
+
+/**
+ * @brief Check if the matrix has any NaN or INF elements.
+ *
+ * @tparam Derived The type of the matrix.
+ * @param M The matrix to check.
+ * @return true if the matrix has any NaN or INF elements, false otherwise.
+ */
+template <typename Derived>
+inline bool is_invalid_matrix(const Eigen::MatrixBase<Derived> &M) {
+  bool has_nan = !(M.array() == M.array()).all();
+  bool has_inf = M.array().isInf().any();
+  return has_nan || has_inf;
+}
+
+/**
+ * @brief Calculates the right pseudoinverse of the given matrix.
+ *
+ * @param M The matrix to calculate the pseudoinverse of.
+ * @throws char* if the pseudoinverse is invalid.
+ * @return The pseudoinverse of the given matrix.
+ */
+inline Eigen::MatrixXd calculate_right_pseudoinverse(const Eigen::MatrixXd &T) {
+  Eigen::MatrixXd pseudoinverse = T.transpose() * (T * T.transpose()).inverse();
+  // pseudoinverse.completeOrthogonalDecomposition().pseudoInverse();
```

Comment:
Remove if not used

Reply:
Is used in the thruster_allocator_ros.cpp 
pseudoinverse_allocator_.T_pinv =
      calculate_right_pseudoinverse(thrust_configuration);

Reply:
Ok nvm i got what you meant